### PR TITLE
Translator refactoring

### PR DESCRIPTION
**Pull request by toasted-nutbread**
Fixes the bug noted in #337. First commit is some refactoring, the second is the fix.

### CODE REVIEW

File: ext/bg/js/dictionary.js [hunk 1]
```diff
@@ -159,11 +159,14 @@ function dictTermsGroup(definitions, dictionaries) {
...
-    const definitionsBySequence = {'-1': []};
+    const sequencedDefinitions = new Map();
+    const nonSequencedDefinitions = [];
     for (const definition of definitions) {
-        if (mainDictionary === definition.dictionary && definition.sequence >= 0) {
-            if (!definitionsBySequence[definition.sequence]) {
-                definitionsBySequence[definition.sequence] = {
+        const sequence = definition.sequence;
+        if (mainDictionary === definition.dictionary && sequence >= 0) {
+            sequencedDefinition = sequencedDefinitions.get(sequence);
```
Comment (siikamiika) [inline ext/bg/js/dictionary.js]: `let`

**Comment by siikamiika** [comment]
I noticed another bug when testing: "Main dictionary for merged mode" is shown by "Active profile" instead of "Modifying profile". If you import a dictionary without `termTags` before one that has them and set `mainDictionary` to the one that has them without doing it for each profile while they're active, you will end up in a situation where the popup has wrong dictionary as the main dictionary. Example of the situation in attached config.

[yomichan-settings-2020-01-29-20-45-03.txt](https://github.com/FooSoft/yomichan/files/4130674/yomichan-settings-2020-01-29-20-45-03.txt)

File: ext/bg/js/dictionary.js (hunk 1 above)
Comment (toasted-nutbread) [reply ext/bg/js/dictionary.js]: I forgot that eslint won't catch that due to `"no-undef": "off"`

**Comment by toasted-nutbread** [comment]
> 
> 
> I noticed another bug when testing: "Main dictionary for merged mode" is shown by "Active profile" instead of "Modifying profile". If you import a dictionary without `termTags` before one that has them and set `mainDictionary` to the one that has them without doing it for each profile while they're active, you will end up in a situation where the popup has wrong dictionary as the main dictionary. Example of the situation in attached config.
> 
> [yomichan-settings-2020-01-29-20-45-03.txt](https://github.com/FooSoft/yomichan/files/4130674/yomichan-settings-2020-01-29-20-45-03.txt)

Fixing this in #343 because it's outside the scope of this change.

**Comment by siikamiika** [comment]
I'm still not sure why #337 originally happened, but the changes look good to me.

**Comment by toasted-nutbread** [comment]
> I'm still not sure why #337 originally happened, but the changes look good to me.

I can't replicate the exact scenario either, but this does handle an issue where the format of sequenced definition data didn't match non-sequenced definitions, despite them all being added to the same results data.

**Pull request merged**